### PR TITLE
Dashboard addition

### PR DESCRIPTION
**Pull request by justas-b**

### CODE REVIEW

File: Dashboard/app.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+     Output(component_id='user-power-id',
+            component_property='children'),
+     Output(component_id="bpm-id", component_property="style")],
+    [Input("interval-component", "n_intervals"), ]
+)
+def run_consumer(n):
+    print('Attempting to read from kafka')
+    reading = read_in_from_kafka()
+    if reading:
+        print("Reading OUTPUT")
```
Comment (ArunMRK) [inline Dashboard/app.py:54]: Maybe add a comment explaining what this is doing?

**Review by ArunMRK — COMMENTED** [review]
Relatively few comments since we actually wrote this code all together in a live share session

File: Dashboard/pages/recent_rides.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+             interval=1000,
+             n_intervals=0
+             ),
+        dcc.Interval(
+            id="riders-interval-component",
+            interval=60*1000,
+            n_intervals=0
+        ),
+        dcc.Interval(
+            id="interval-component",
```
Comment (bendg4444) [inline Dashboard/pages/recent_rides.py:25]: Rename this to be more specific, I believe it links to the get kafkha data callback

File: Dashboard/pages/recent_rides.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        # Title / top div
+        html.Div([
+            # title wrapper div
+            html.Div([
+                html.H2(children="Current Rider: ")
+            ], style={"display": "inline-block", "width": "20%"}),
+            # div for last updated text
+            html.Div([
+                html.Div([
+                    # last updated text ("last updated 30 seconds ago..")
```
Comment (bendg4444) [inline Dashboard/pages/recent_rides.py:40]: This comment seems redudent or not in the correct place

File: Dashboard/pages/recent_rides.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+                        id="bpm-id", style={"text-align": "center", "font-weight": "bold", "font-size": "20px"}),
+                ], style={"display": "inline-block", "width": "33%"}),
+                html.Div([
+                    # Total Power
+                    html.Div(id="max-heart-rate-id",
+                             style={"text-align": "center", "font-size": "20px"}),
+                ], style={"display": "inline-block", "width": "33%"}),
+            ], style={"display": "inline-block", "width": "50%"}),
+
+            # Wrapper for max / current
```
Comment (bendg4444) [inline Dashboard/pages/recent_rides.py:106]: We removed Max/Current so its not just a wrapper

File: Dashboard/pages/recent_rides.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    ]),
+
+    # Age aggregation graph div
+    html.Div([
+        # age aggregation graph div
+        html.Div(id="age",
+                    style={"text-align": "center", "font-size": "30px", "padding-left": "100px", "padding-right": "100px"})
+    ]),
+    dcc.Interval(
+        id="interval-component",
```
Comment (bendg4444) [inline Dashboard/pages/recent_rides.py:188]: This interval component is redudent, we have them stored at the top

File: Dashboard/read_in_from_kafka.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
+from kafka_consumer import *
+from current_ride_utils import *
+import pprint
+
+def read_in_from_kafka():
+
+    # if found_details == None:
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:7]: Remove commented out code

File: Dashboard/read_in_from_kafka.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                "datetime": None, "duration": None, "current_resistance": 0,
+                "current_heart_rate": 0, "current_rpm": 0, "current_power": 0,
+                "max_resistance": 0,"max_rpm": 0, "max_power": 0,
+                "total_power": 0
+            }
+            heart_rate_abnormal=False
+            max_user_heart_rate = 0
+
+        elif 'user_id' in message:
+            '''New user found'''
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:44]: Use """ instead of '''

File: Dashboard/read_in_from_kafka.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                
+        # elif found_user and ("Telemetry - hrt" in message):
+        elif ("Telemetry - hrt" in message):
+            print('Kafka reader line 69')
+            '''  get second part of data if user exists'''
+            ride_hrt_rpm_power = extract_ride_hrt_rpm_power(message)
+            heart_rate = int(ride_hrt_rpm_power["heart_rate"])
+            print('line 73')
+            # if (heart_rate > max_heart_rate):
+            #     '''Trigger that checks the heart rate'''
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:75]: Remove commented out code

File: Dashboard/read_in_from_kafka.py [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            #     heart_rate_abnormal=True
+
+            print('line 78')
+            current_ride_data["current_heart_rate"] = heart_rate
+            current_ride_data["current_rpm"] = int(ride_hrt_rpm_power["rpm"])
+            current_ride_data["current_power"] = float(ride_hrt_rpm_power["power"])
+            current_ride_data["total_power"] += float(ride_hrt_rpm_power["power"])
+            print('line 83')
+            # if current_ride_data["max_rpm"] < int(ride_duration_resistance["rpm"]):
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:84]: Remove commented out code

File: Dashboard/read_in_from_kafka.py [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            current_ride_data["current_resistance"] = int(ride_duration_resistance["resistance"])
+            current_ride_data["datetime"]= ride_duration_resistance["date_time"]
+
+            if current_ride_data["max_resistance"] < int(ride_duration_resistance["resistance"]):
+                current_ride_data["max_resistance"] = int(ride_duration_resistance["resistance"])
+                
+        # elif found_user and ("Telemetry - hrt" in message):
+        elif ("Telemetry - hrt" in message):
+            print('Kafka reader line 69')
+            '''  get second part of data if user exists'''
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:70]: """ instead of '''

File: Dashboard/read_in_from_kafka.py [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+            current_ride_data["current_heart_rate"] = heart_rate
+            current_ride_data["current_rpm"] = int(ride_hrt_rpm_power["rpm"])
+            current_ride_data["current_power"] = float(ride_hrt_rpm_power["power"])
+            current_ride_data["total_power"] += float(ride_hrt_rpm_power["power"])
+            print('line 83')
+            # if current_ride_data["max_rpm"] < int(ride_duration_resistance["rpm"]):
+            #     current_ride_data["max_rpm"] = int(ride_duration_resistance["rpm"])
+            # if current_ride_data["max_power"] < int(ride_duration_resistance["power"]):
+            #     current_ride_data["max_power"] = int(ride_duration_resistance["power"])
+            print('line 88')
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:88]: redundent now

File: Dashboard/read_in_from_kafka.py [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+        
+    # print('reached 1')
```
Comment (bendg4444) [inline Dashboard/read_in_from_kafka.py:90]: remove commented out code

**Review by bendg4444 — CHANGES_REQUESTED** [review]
Please review the changes

**Pull request merged**